echopype/utils/prov.py

```diff
@@ -0,0 +1,31 @@
+from datetime import datetime as dt
+from typing import Dict
+
+from _echopype_version import version as ECHOPYPE_VERSION
+from typing_extensions import Literal
+
+ProcessType = Literal["conversion", "processing"]
+
+
+def echopype_prov_attrs(process_type: ProcessType, source_files: str = None) -> Dict[str, str]:
+    """
+    Standard echopype software attributes for provenance
+
+    Parameters
+    ----------
+    process_type : ProcessType
+        Echopype process function type
+    source_files: str
+        Source file path. A list of files is not currently supported
+    """
+    prov_dict = {
+        f"{process_type}_software_name": "echopype",
+        f"{process_type}_software_version": ECHOPYPE_VERSION,
+        f"{process_type}_time": dt.utcnow().isoformat(timespec="seconds") + "Z",  # use UTC time
+    }
+    if source_files:
+        # TODO: src_filenames will be replaced with a new variable, source_filenames
+        #   Also, come to think of it, source files is not "echopype provenance" info per se
```

Comment:
What I meant with "source files is not "echopype provenance" info per se"  was that all the other attributes I've included in `echopype_prov_attrs` are about the echopype software itself, but source files is not. Though, come to think of it, strictly speaking, `conversion_time` is not either .... Hmm. We could rename `echopype_prov_attrs` to something more general. Its scope is the software used (name & version) and basic info about its application (timestamp when it was run, and input files used -- though the latter will be moved to a variable and will no longer be an attribute)

Reply:
Sure, sounds good. It sounds like you're also saying that we keep the source_files provenance functionality in this function, even though it'll be changed to add xarray variables rather than an attribute. Correct?

Reply:
> It sounds like you're also saying that we keep the source_files provenance functionality in this function, even though it'll be changed to add xarray variables rather than an attribute. Correct?

I didn't think that far... ha. I think you are right that since it will be a data variable not an attribute, it should not be here. In that case, the function name would become `echopype_ver_time_prov`.